addons/account/models/account_move.py

```diff
@@ -4418,12 +4418,21 @@ def _add_cash_basis_lines_to_exchange_difference_vals(lines, exchange_diff_move_
                     if line.tax_repartition_line_id:
                         # Tax line.
                         grouping_key = self.env['account.partial.reconcile']._get_cash_basis_tax_line_grouping_key_from_record(line)
-                        account_vals_to_fix[grouping_key] = {
-                            **vals,
-                            'account_id': line.account_id.id,
-                            'tax_base_amount': line.tax_base_amount,
-                            'tax_repartition_line_id': line.tax_repartition_line_id.id,
-                        }
+                        if grouping_key in account_vals_to_fix:
+                            balance = account_vals_to_fix[grouping_key]['debit'] - account_vals_to_fix[grouping_key]['credit']
+                            balance += vals['debit'] - vals['credit']
+                            account_vals_to_fix[grouping_key].update({
+                                'debit': balance if balance > 0.0 else 0.0,
+                                'credit': -balance if balance < 0.0 else 0.0,
```

Comment:
Why not simply 

`
'debit': account_vals_to_fix[grouping_key]['debit'] + vals['debit'],

'credit': account_vals_to_fix[grouping_key]['credit'] + vals['credit']
`

instead of doing it with the balance ?